index.js

```diff
@@ -138,6 +149,270 @@ app.intent('send.message - no', async conv => {
+  });
+  return;
+}) 
+
+
+/**
+ * join.conference - collect.target
+ */
+app.intent('join.conference - collect.target', async conv => {
+  const circuit = await getCircuit(conv);
+  if (!circuit) {
+    return;
+  }
+
+  const { calls, titles } = conv.contexts.input['joinconference_gettarget'].parameters;
+  const { target } = conv.parameters;
+  let callId = null;
+
+  // Lowercase titles so if the name of the conference is Test and they type in test
+  // it will still be found. That is the reason for the lowercasing the target and titles.
+  const lcTitles = titles.map(title => title.toLowerCase());
+  const index = lcTitles.indexOf(target);
+
+  if (index !== -1) {
+    callId = calls[index].callId;
+  } else {
+    for (let call of calls) {
+      if (call.title.toLowerCase() === target.toLowerCase())  {
+        callId = call.callId;
+      } 
+    }
+  }
+
+  conv.ask(`<speak>Ready to join the <break time="0.5s"/>${target}<break time="0.5s"/> conference call?</speak>`, new Suggestions('Yes', `No, don't join`));
+  conv.contexts.set('joinconference_send', 5, {
+    callId: callId
+  });
+}); 
+
+/**
+ * join.conference - yes
+ */
+app.intent('join.conference - yes', async conv => {
+  const circuit = await getCircuit(conv);
+  if (!circuit) {
+    return;
+  }
+  const { clientId } = await findWebClient(circuit);
+  const { callId } = conv.contexts.input['joinconference_send'].parameters;
+  await circuit.joinConference(callId, {audio: true, video: false}, clientId);
+  conv.contexts.delete('joinconference_data');
+  conv.ask('Conference joined. Is there anything else I can do for you?');
+  conv.ask(new Suggestions('No, that\'s all', 'Yes'));
+  conv.contexts.set('anything_else', 2);
+}); 
+
+/**
+ * join.conference - no
+ */
+app.intent('join.conference - no', async conv => {
+  conv.contexts.delete('joinconference_data');
+  conv.ask('Conference not joined. Is there anything else I can do for you?');
+  conv.ask(new Suggestions('No, that\'s all', 'Yes'));
+  conv.contexts.set('anything_else', 2);
+});
+
+/**
+ * leave.conference
+ */
+app.intent('leave.conference', async (conv, {target}) => {
+  const circuit = await getCircuit(conv);
+  if (!circuit) {
+    return;
+  }
+
+  let { inConferenceCall, calls } = await getConferenceInfo(conv);
+ 
+  if (!inConferenceCall) {
+    conv.contexts.delete('joinconference_data');
+    conv.ask("<speak>You aren't in any conference calls. Is there anything else I can do for you?</speak>");
+    conv.ask(new Suggestions('No, that\'s all', 'Yes'));
+    conv.contexts.set('anything_else', 2); 
+    return;
+  }
+
+  let titles = [];
+  calls.forEach((call) => {
+    if (call.title === "") {
+      let placeholder = truncate(call.topicPlaceholder);
+      titles.push(placeholder.toLowerCase());
+    } else {
+      titles.push(call.title.toLowerCase());
+    }
+  });
+
+  // If not undefined just send them to leaveconference_send
+  if (target) {
+    let i = titles.indexOf(target.toLowerCase());
```

Comment:
const, I will stop putting this but go through the code and anywhere you can use const instead of let please change it.